extensions-core/kafka-extraction-namespace/src/main/java/io/druid/query/lookup/KafkaLookupExtractorFactory.java

```diff
@@ -79,10 +76,12 @@ public String fromBytes(byte[] bytes)
   private final AtomicLong doubleEventCount = new AtomicLong(0L);
   private final NamespaceExtractionCacheManager cacheManager;
   private final String factoryId = UUID.randomUUID().toString();
-  private final AtomicReference<Map<String, String>> mapRef = new AtomicReference<>(null);
-  private final AtomicBoolean started = new AtomicBoolean(false);
+  private final ConsumerConnectorFactory consumerConnectorFactory;
+  private final AtomicReference<State> currentState = new AtomicReference<>(State.NEW);
+  private final CountDownLatch startingReads = new CountDownLatch(1);
```

Comment:
This is yet another tangled piece of lifecycle that should be eliminated. I have not been able to do this, see the comments to where its `countDown` is invoked.